source/code/go/src/plugins/oms.go

```diff
@@ -0,0 +1,263 @@
+	// NameIDMap caches the container it to Name mapping
+	NameIDMap map[string]string
+	// IgnoreIDSet set of  container Ids of kube-system pods
+	IgnoreIDSet map[string]bool
+)
+
+var (
+	// FLBLogger stream
+	FLBLogger = createLogger()
+	// Log method
+	Log = FLBLogger.Printf
+)
+
+const containerInventoryPath = "/var/opt/microsoft/docker-cimprov/state/ContainerInventory"
+
+// ContainerInventory represents the container info
+type ContainerInventory struct {
+	ElementName       string `json:"ElementName"`
+	CreatedTime       string `json:"CreatedTime"`
+	State             string `json:"State"`
+	ExitCode          int    `json:"ExitCode"`
+	StartedTime       string `json:"StartedTime"`
+	FinishedTime      string `json:"FinishedTime"`
+	ImageID           string `json:"ImageId"`
+	Image             string `json:"Image"`
+	Repository        string `json:"Repository"`
+	ImageTag          string `json:"ImageTag"`
+	ComposeGroup      string `json:"ComposeGroup"`
+	ContainerHostname string `json:"ContainerHostname"`
+	Computer          string `json:"Computer"`
+	Command           string `json:"Command"`
+	EnvironmentVar    string `json:"EnvironmentVar"`
+	Ports             string `json:"Ports"`
+	Links             string `json:"Links"`
+}
+
+// DataItem represents the object corresponding to the json that is sent by fluentbit tail plugin
+type DataItem struct {
+	LogEntry          string `json:"LogEntry"`
+	LogEntrySource    string `json:"LogEntrySource"`
+	LogEntryTimeStamp string `json:"LogEntryTimeStamp"`
+	ID                string `json:"Id"`
+	Image             string `json:"Image"`
+	Name              string `json:"Name"`
+	SourceSystem      string `json:"SourceSystem"`
+	Computer          string `json:"Computer"`
+}
+
+// ContainerLogBlob represents the object corresponding to the payload that is sent to the ODS end point
+type ContainerLogBlob struct {
+	DataType  string     `json:"DataType"`
+	IPName    string     `json:"IPName"`
+	DataItems []DataItem `json:"DataItems"`
+}
+
+func populateMaps() {
+	files, err := ioutil.ReadDir(containerInventoryPath)
+
+	if err != nil {
+		Log("error when reading container inventory")
+		log.Fatal(err.Error())
+	}
+
+	for _, file := range files {
+		fullPath := fmt.Sprintf("%s/%s", containerInventoryPath, file.Name())
+		fileContent, err := ioutil.ReadFile(fullPath)
+		if err != nil {
+			Log("Error reading file content %s", fullPath)
+			log.Fatal(err)
+		}
+		var containerInventory ContainerInventory
+		unmarshallErr := json.Unmarshal(fileContent, &containerInventory)
+
+		if unmarshallErr != nil {
+			Log("Unmarshall error when reading file %s", fullPath)
+			log.Fatal(unmarshallErr)
+		}
+
+		ImageIDMap[file.Name()] = containerInventory.Image
+		NameIDMap[file.Name()] = containerInventory.ElementName
+	}
+}
+
+func createLogger() *log.Logger {
+	logfile, err := os.Create(filepath.Join("/shared/data", fmt.Sprintf("fluent-bit-runtime_%s.log", time.Now().Format("2006-01-02T15-04-05"))))

```

Comment:
can we change this to our docker provider folder ? Also you need to drop an empty file chmod on the file to be able to create & write a new log file.